pkg/agent/route/route_linux.go

```diff
@@ -397,6 +432,9 @@ func (c *Client) Initialize(nodeConfig *config.NodeConfig, done func()) error {
 	if c.nodeLatencyMonitorEnabled {
 		c.initNodeLatencyRules()
 	}
+	if c.proxyAll && c.proxyHealthCheckPort != 0 {
```

Comment:
I noticed that !serviceHealthServerDisabled was also checked in https://github.com/antrea-io/antrea/pull/7371, it seems we also need to check it as well before install the rules?